From: Peter Hurley <peter@hurleysoftware.com>
Date: Sat, 16 Jan 2016 15:23:38 -0800
Subject: of: earlycon: of_setup_earlycon() requires CONFIG_OF_EARLY_FLATTREE
Git-commit: 8477614d9f7c5ce0e871e1aa200d9060a6b8749e
Patch-mainline: v4.6-rc1
References: fate#322061

DT earlycon is only supported for CONFIG_OF_EARLY_FLATTREE=y; exclude
of_setup_earlycon() if not defined.

Signed-off-by: Peter Hurley <peter@hurleysoftware.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/tty/serial/earlycon.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/tty/serial/earlycon.c
+++ b/drivers/tty/serial/earlycon.c
@@ -211,6 +211,8 @@ static int __init param_setup_earlycon(c
 }
 early_param("earlycon", param_setup_earlycon);
 
+#ifdef CONFIG_OF_EARLY_FLATTREE
+
 int __init of_setup_earlycon(unsigned long addr,
 			     int (*setup)(struct earlycon_device *, const char *))
 {
@@ -271,3 +273,5 @@ int __init of_setup_earlycon(const struc
 	register_console(early_console_dev.con);
 	return 0;
 }
+
+#endif /* CONFIG_OF_EARLY_FLATTREE */
